### ISSUE TITLE
QtWebKit+GStreamer=0 features

### ISSUE THREAD

**Issue by ZicoRio** [annulen/webkit#813]
I've compiled QtWebKit+GStreamer.
Environment:
OS: Windows 7
VS: 2015 mode x64
Patches: https://github.com/annulen/webkit/issues/698
GStreamer: https://gstreamer.freedesktop.org/data/pkg/windows/1.16.0/
pkg-config: 0.28-1
Result: 0 errors
Open site: https://www.youtube.com/html5
Result: 0 features:
No HTMLVideoElement (!)
No H.264 (x)
No WebM VP8 (x)
No Media Source Extensions (x)
No MSE & H.264 (x)
No MSE & WebM VP9 (x)
------------
Why !?
------------
[CMakeCache.txt](https://github.com/annulen/webkit/files/3183735/CMakeCache.txt)
------------
Settings:
-DENABLE_API_TEST=OFF -DUSE_GSTREAMER=ON -DUSE_GSTREAMER_DEFAULT=ON -DUSE_MEDIA_FOUNDATION=OFF -DENABLE_MEDIA_SOURCE=ON -DENABLE_VIDEO=ON -DENABLE_WEB_AUDIO=ON -DUSE_GSTREAMER_MPEGTS=ON -DUSE_GSTREAMER_GL=ON

**Comment by annulen** [comment]
Could you attach `cmakeconfig.h`?

**Comment by ZicoRio** [comment]
cmakeconfig.h
[cmakeconfig.h.txt](https://github.com/annulen/webkit/files/3183887/cmakeconfig.h.txt)

**Comment by annulen** [comment]
It looks good. Make sure that gstreamer can load its plugins when your application is running. Also, please check this page: http://camendesign.com/code/video_for_everybody/test.html

**Comment by ZicoRio** [comment]
I used ProcessMonitor - all requested files are available.
http://camendesign.com/code/video_for_everybody/test.html
Yes this video plays via Flash player.

**Comment by annulen** [comment]
Weird. By chance, haven't you disabled `QWebSettings::MediaEnabled` in your code?

**Comment by ZicoRio** [comment]
QWebSettings::MediaEnabled
QWebSettings::MediaSourceEnabled
All enabled.

**Comment by annulen** [comment]
Then I'm out of ideas

**Comment by ZicoRio** [comment]
I use last versions of libraries:
zlib, libxml2, icu59, ...
Maybe this is mistake ?

**Comment by annulen** [comment]
No

**Comment by ZicoRio** [comment]
I compiled under VS2015, GStreamer DLLs - VS2017
Maybe this is mistake ?

**Comment by annulen** [comment]
They are binary compatible, and if they weren't your process would crash much earlier

**Comment by annulen** [comment]
If you type `HTMLVideoElement` in Inspector's console, what does it print? (Should be function with native code with your configuration)

**Comment by ZicoRio** [comment]
In Inspector:
<object width="640" height="360" type="application/x-shockwave-flash" data="player.swf">

<param name="movie" value="player.swf">
<param name="flashvars" value="autostart=true&amp;controlbar=over&amp;image=poster.jpg&amp;file=http://clips.vorwaerts-gmbh.de/VfE_flash.mp4">

<img src="poster.jpg" width="640" height="360" alt="Big Buck Bunny" title="No video playback capabilities, please download the video below">
</object>
By the way - same result for GStreamer v1.12.4
I don't know that to think...

**Comment by annulen** [comment]
I mean: what result do you see if you type `HTMLVideoElement` into inspector's input propmt? Also, do you see any JS errors in "Console" tab?

**Comment by annulen** [comment]
GStreamer version is probably irrelevant - any version starting from1.2.x is fine if WebKit compiles with it, and it has necessary codecs

**Comment by annulen** [comment]
IIRC @Emdek is using GStreamer with MinGW (MXE) and it works fine

**Comment by Emdek** [comment]
Yep, I have a GStreamer enabled build that works even with Windows XP.
